templates/tfengine/components/project/variables.tf

```diff
@@ -0,0 +1,98 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+{{$props := .__schema__.properties -}}
+{{$projectProps := $props.project.properties -}}
+{{$apiIdentitiesProps := $projectProps.api_identities.items.properties -}}
+{{$sharedVpcProps := $projectProps.shared_vpc_attachment.properties -}}
+variable "billing_account" {
+  type        = string
+  description = {{schemaDescription $props.billing_account.description}}
+}
+
+variable "parent_id" {
+  type        = string
+  description = {{schemaDescription $props.parent_id.description}}
+  validation {
+    condition     = can(regex("{{$props.parent_id.pattern}}", var.parent_id))
+    error_message = "The parent_id must be valid. Should have only numeric values with a length between 8 and 25 digits. See https://cloud.google.com/resource-manager/docs/cloud-platform-resource-hierarchy to know how to get your organization/folder id."
+  }
+}
+
+
+variable "parent_type" {
+  type        = string
+  description = {{schemaDescription $props.parent_type.description}}
+  validation {
+    condition     = can(regex("{{$props.parent_type.pattern}}", var.parent_type))
+    error_message = "The parent_type must be valid. Should have only numeric values with a length between 8 and 25 digits. See https://cloud.google.com/resource-manager/docs/cloud-platform-resource-hierarchy to know how to get your organization/folder id."
+  }
+}
+
+variable "api_identities" {
+  type = list(object({
+    api   = string
+    roles = list(string)
+  }))
+  description = <<EOF
+    {{$projectProps.api_identities.description}}
+
+    * api = {{$apiIdentitiesProps.api.description}}
```

Comment:
Same comment as in the other PRs, use range and add a "Fields" header.

Reply:
Done